lib/tasks/subscriptions.rake

```diff
@@ -7,4 +7,34 @@ namespace :subscriptions do
       subscription.update!(unique_id: subscription.customer.customer_id)
     end
   end
+
+  # NOTE: Ability to create invoices in the future.
+  # How to use it: bundle exec rake "subscriptions:generate_invoice[timestamp, external_id1, external_id2, ...]"
+  # ie bundle exec rake "subscriptions:generate_invoice[11675267200, 3679fa0c-d0fd-47a7-8954-a5115762e544]"
+  desc 'Generate invoice for a specific timestamp'
+  task :generate_invoice, [:timestamp] => :environment do |_task, args|
+    abort "Missing timestamp and external subscription ids\n\n" unless args[:timestamp]
+    abort "Missing external subscription ids\n\n" if args.extras.blank?
+
+    subscriptions = Subscription.where(external_id: args.extras)
+
+    abort "External subscription ids not found\n\n" if subscriptions.blank?
+    abort "Subscriptions don't belong to the same customer\n\n" if subscriptions.pluck(:customer_id).uniq.count > 1
+
+    result = Invoices::SubscriptionService.new(subscriptions:, timestamp: args[:timestamp].to_i, recurring: false).create
+    invoice = result.invoice
+
+    # NOTE: Do not generate the PDF file if invoice is draft.
+    return if invoice.draft?
+
+    pdf_service = Utils::PdfGenerator.new(template: 'invoice', context: invoice)
```

Comment:
Just a comment: to prevent code duplication, maybe we could call the `generate_pdf` private method of the `Invoices::GenerateService` to generate this invoice. What do you think?

Reply:
Yeah, good spot 👍

Reply:
We probably can directly call `Invoices::GenerateService.generate(invoice_id: invoice.id)`